src/Dash.NET.ComponentGeneration/DocumentationGeneration.fs

```diff
@@ -0,0 +1,198 @@
+﻿module Dash.NET.ComponentGeneration.DocumentationGeneration
+
+open ReactMetadata
+open Prelude
+
+let getIsRequired (req: bool) =
+    if req then
+        "required"
+    else
+        "optional"
+
+let getTypeIsRequired (ptype: SafeReactPropType) =
+    (ptype |> SafeReactPropType.getProps).required
+    |> Option.defaultValue false 
+    |> getIsRequired
+
+let getTypeDescription (ptype: SafeReactPropType) =
+    (ptype |> SafeReactPropType.getProps).description
+
+let getTypeHasDefault (dval: SafeReactPropType) =
+    match dval with
+    | Any (_, Some v) -> sprintf "default %s" v |> Some
+    | _ -> None
+
+//TODO nicer formatting for complex types
+let rec getTypePropDocumentation (ptype: SafeReactPropType) =
+    match ptype with 
+    | Array _ -> "list" |> Some
+    | Bool _ -> "boolean" |> Some
+    | Number _ -> "number" |> Some
+    | String _ -> "string" |> Some
+    | Object _ -> "record" |> Some
+    | Any _ -> "boolean | number | string | record | list" |> Some
+    | Element _ -> "dash component" |> Some
+    | Node _ -> "a list of or a singular dash component, string or number" |> Some
+
+    // Special cases, each type will have a unique name
+    | Enum (_, Some values) -> 
+        let valueList =
+            values
+            |> List.choose (fun case -> 
+                match case with 
+                | Any (_, Some value) -> value |> Some
+                | _ -> None)
+        
+        if valueList.Length > 1 then
+            valueList
+            |> List.map (sprintf "'%s'")
+            |> List.reduce (sprintf "%s, %s")
+            |> sprintf "value equal to: %s"
+            |> Some
+        else if valueList.Length = 1 then
+            sprintf "value equal to: '%s'" valueList.[0]
+            |> Some
+        else
+            sprintf "value equal to: unknown"
+            |> Some
+
+    | Union (_, Some values) ->
+        let valueList =
+            values
+            |> List.choose getTypePropDocumentation
+        
+        if valueList.Length > 1 then
+            valueList
+            |> List.reduce (sprintf "%s | %s")
+            |> Some
+        else if valueList.Length = 1 then
+            valueList.[0]
+            |> Some
+        else
+            sprintf "unknown"
+            |> Some
+
+    | ArrayOf (_, Some value) -> 
+        value
+        |> getTypePropDocumentation
+        |> Option.map (sprintf "list with values of type: %s")
+
+    | ObjectOf (_, Some value) -> 
+        value
+        |> getTypePropDocumentation
+        |> Option.map (sprintf "dict with values of type: %s")
```

Comment:
dict -> record?

(apologies for the late comment, just poking around here to see how you implemented things)

Reply:
not in this case. `ObjectOf` is actually a `Map`, which is the F# version of `IDictionary`. I think I put `dict` here to be consistent with the python generator, but this technically should be `Map`.

Reply:
Got it - just wanted to make sure this wasn't an unintended remnant from Python :)